baseplate/lib/prometheus_metrics.py

```diff
@@ -26,316 +26,348 @@
+    prefix = "http_client"
+
+    # http client labels and metrics
+    latency_labels = [
+        "http_method",
+        "http_success",
+        "http_slug",
+    ]
+    requests_total_labels = [
+        "http_method",
+        "http_success",
+        "http_response_code",
+        "http_slug",
+    ]
+    active_requests_labels = [
+        "http_method",
+        "http_slug",
+    ]
+
+    # Latency histogram of HTTP calls made by clients
+    # buckets are defined above (from 100µs to ~14.9s)
+    latency_seconds = Histogram(
+        f"{prefix}_latency_seconds",
+        "Latency histogram of HTTP calls made by clients",
+        latency_labels,
+        buckets=default_latency_buckets,
+    )
+    # Counter counting total HTTP requests started by a given client
+    requests_total = Counter(
+        f"{prefix}_requests_total",
+        "Total number of HTTP requests started by a given client",
+        requests_total_labels,
+    )
+    # Gauge showing current number of active requests by a given client
+    active_requests = Gauge(
+        f"{prefix}_active_requests",
+        "Number of active requests for a given client",
+        active_requests_labels,
+    )
+
     def __init__(self) -> None:
         pass
 
-    def latency_seconds_metric(self, tags: Dict[str, Any]) -> Histogram:
-        return http_client_latency_seconds.labels(
+    @classmethod
+    def latency_seconds_metric(cls, tags: Dict[str, Any]) -> Histogram:
+        return cls.latency_seconds.labels(
             http_method=tags.get("http.method", ""),
             http_success=getHTTPSuccessLabel(int(tags.get("http.status_code", "0"))),
             http_slug=tags.get("http.slug", ""),
         )
 
-    def requests_total_metric(self, tags: Dict[str, Any]) -> Counter:
-        return http_client_requests_total.labels(
+    @classmethod
+    def requests_total_metric(cls, tags: Dict[str, Any]) -> Counter:
+        return cls.requests_total.labels(
             http_method=tags.get("http.method", ""),
             http_success=getHTTPSuccessLabel(int(tags.get("http.status_code", "0"))),
             http_response_code=tags.get("http.status_code", ""),
             http_slug=tags.get("http.slug", ""),
         )
 
-    def active_requests_metric(self, tags: Dict[str, Any]) -> Gauge:
-        return http_client_active_requests.labels(
+    @classmethod
+    def active_requests_metric(cls, tags: Dict[str, Any]) -> Gauge:
+        return cls.active_requests.labels(
             http_method=tags.get("http.method", ""),
             http_slug=tags.get("http.slug", ""),
         )
 
-    def get_latency_seconds_metric(self) -> Histogram:
-        return http_client_latency_seconds
+    @classmethod
+    def get_latency_seconds_metric(cls) -> Histogram:
+        return cls.latency_seconds
 
-    def get_requests_total_metric(self) -> Counter:
-        return http_client_requests_total
+    @classmethod
+    def get_requests_total_metric(cls) -> Counter:
```

Comment:
Won't this share the gauge and counter across spans if they're class methods?

Reply:
They are shared. The tags is the only thing differencing these across spans. The underlying object is the same and a singleton. We only ever created one of them outside of the class to begin with.